email_service/utils.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import os
+from email.mime.image import MIMEImage
+
+import html2text
+from django.conf import settings
+from django.core.files.base import ContentFile
+from django.core.mail.message import EmailMultiAlternatives
+from django.template.loader import render_to_string
+from jinja2 import Template as JinjaTemplate
+
+logger = logging.getLogger("emails")
+
+
+def send_custom_email(
+    recipient: list[str] | str,
+    path: str = None,
+    template: any = None,
+    template_prefix: str = None,
+    context=None,
+    subject: str = None,
+    body: str = None,
+    attachments: list[ContentFile] = None,
+    attachment_path: str = None,
+    enable_logo: bool = False,
+) -> None:
+    """This function is responsible to send emails and create emails in database model.
+
+    Parameters:
+    recipient (list[str]): List of Receivers emails
+    path (str): path to html file for email content and subject (path should be upto parent folder of html file inside template folder)
+    template (any): Object of Template Model if exists (Optional)
+    template_prefix (str): html file name
+    context (dict): context to replace variable name in template
+    subject (str): subject of email as string
+    body (str): content of email as string
+    attachment (list[ContentFile]): attach file to send email as attachement
+    attachment_path (str): send string path of attachment,
+    enable_logo (bool): set true to enable logo in email
+
+    Returns:
+    str : email success or error message
+    """
+    from email_service.models import Attachment, Email
+
+    if not recipient or len(recipient) == 0:
+        logger.error("Please provide at least one recipient.")
+        return "Please provide at least one recipient."
+    if (path or template_prefix) and (subject or body):
+        return (
+            "You can either send templated email or simple email at a time, not both."
+        )
+    if not ((path and template_prefix) or subject):
+        logger.error(
+            "Please provide either path to html template or text subject of email."
+        )
+        return "Please provide either path to html template or text subject of email."
+
+    if not ((path and template_prefix) or body):
+        logger.error(
+            "Please provide either path to html template or text body of email."
+        )
+        return "Please provide either path to html template or text body of email."
+
+    try:
```

Comment:
remove try block use if else statement if we want to do defensive coding technique.